File: 53-maximum-subarray/53-maximum-subarray.go
func maxSubArray(nums []int) int {
  ans := -10001
  sum := 0
  for _, v := range nums {
    if sum > 0 {
      sum += v
    } else {
      sum = v
    }
    if sum > ans {
      ans = sum
    }
  }
  return ans
}

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
  public int maxSubArray(int[] nums) {
    int ans = -10001;
    int sum = 0;
    for (int n : nums) {
      sum = (sum <= 0) ? n : sum + n;
      ans = Math.max(ans, sum);      
    }
    return ans;
  }
}